compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -690,13 +691,153 @@ FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
 
   return result;
 }
+/// Returns the position of given dimension.
+/// For dim=2, map=affine_map<(d0, d1, d2) -> (d0, d2), it returns 1
+static Optional<unsigned> getResultPosition(AffineMap &map, unsigned dim) {
+  unsigned pos = 0;
+  for (AffineExpr result : map.getResults()) {
+    if (result.cast<AffineDimExpr>().getPosition() == dim) return pos;
+    pos++;
+  }
+  return llvm::None;
+}
+
+/// Finds contiguous loop ranges from the given linalg.generic.
+static SmallVector<ReassociationIndices> getContiguousLoops(
+    linalg::GenericOp genericOp, ArrayRef<unsigned> loopDims) {
+  SmallVector<ReassociationIndices> contiguousLoops;
+  if (loopDims.size() < 2) return contiguousLoops;
+
+  auto areDimensionsContiguous = [&](unsigned preDim, unsigned nextDim) {
+    return llvm::all_of(genericOp.getIndexingMapsArray(), [&](AffineMap map) {
```

Comment:
Nit: Might be easier to assign the `nextPos` to `prePos` after each iteration. I dont know off the top of my head how to do this using `llvm::all_of` but easy to do with a loop.